Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -61,14 +66,109 @@ - (BOOL)installWithOptions:(nonnull SentryOptions *)options
 
     [SentryGlobalEventProcessor.shared
         addEventProcessor:^SentryEvent *_Nullable(SentryEvent *_Nonnull event) {
-            [self.sessionReplay captureReplayForEvent:event];
-
+            if (event.isCrashEvent) {
+                [self resumePreviousSessionReplay:event];
+            } else {
+                [self.sessionReplay captureReplayForEvent:event];
+            }
             return event;
         }];
 
     return YES;
 }
 
+- (void)resumePreviousSessionReplay:(SentryEvent *)event
+{
+    NSURL *dir = [self replayDirectory];
+    NSData *lastReplay =
+        [NSData dataWithContentsOfURL:[dir URLByAppendingPathComponent:@"lastreplay"]];
+    if (lastReplay == nil) {
+        return;
+    }
+
+    NSDictionary<NSString *, id> *jsonObject = [NSJSONSerialization JSONObjectWithData:lastReplay
+                                                                               options:0
+                                                                                 error:nil];
```

Comment:
`SentrySerialization.dataWithJSONObject` transform an object into a NSData.

Reply:
Sorry, I picked the wrong one. Would you mind using `SentrySerialization.deserializeEventEnvelopeItem` and rename that to something more generic since it really has nothing to do with events or envelopes specifically?